src/Systmm.f90

```diff
@@ -83,97 +81,97 @@ SUBROUTINE SYSTMM(temp_file,param_file)
 T_smth=4.0
 
 !
-!     open the output file
+!     open the output file
+!
+
+open(20,file=TRIM(temp_file),status='unknown')
 !
-
-open(20,file=TRIM(temp_file),status='unknown')
-!
 !
 ! Initialize dummy counters that facilitate updating simulated values
-!
-n1=1
-n2=2
-nobs=0
+!
+n1=1
+n2=2
+nobs=0
 ndays=0
 xwpd=nwpd
-hpd=1./xwpd
-!
-!     Year loop starts
-!
-do nyear=start_year,end_year
-  write(*,*) ' Simulation Year - ',nyear,start_year,end_year
-  nd_year=365
-  if (mod(nyear,4).eq.0) nd_year=366
-!
-!     Day loop starts
-!
-  DO nd=1,nd_year
-    year=nyear
-    xd=nd
-    xd_year=nd_year
-!     Start the numbers of days-to-date counter
-!
-    ndays=ndays+1
-!
-!    Daily period loop starts
-!
+hpd=1./xwpd
+!
+!     Year loop starts
+!
+do nyear=start_year,end_year
+  write(*,*) ' Simulation Year - ',nyear,start_year,end_year
+  nd_year=365
+  if (mod(nyear,4).eq.0) nd_year=366
```

Comment:
this is not general enough. 1900 for example is not a leap year. Would be better to abstract this into a separate function `isleapyear(year)` which returns a `true` or `false` and then make sure that `isleapyear()` can handle all cases (there are many examples of correct leap year algorithms out there).